Opaka lengths

MOshima-PIFSC opened the pull request:
<!---
Thanks for opening a PR. This commented text will **NOT** appear in the final PR. Toggle between Write and Preview to see what your PR will look like without the comments.
-->

# What is the feature?
* Updating opaka case study incorporating length data and wrapper functions

# How have you implemented the solution?
* Completely overhauled the code  
* Also updated the `get_ss3_data()` function

# Does the PR impact any other area of the project, maybe another repo?
* Updated `get_ss3_data()` function may impact other case studies that use it

kellijohnson-NOAA commented:
Thank you @MOshima-PIFSC for the PR. I will look at this on Thursday the 16th.

kellijohnson-NOAA commented on content/opaka_dat_fims.RData:
I see that this file is only called and saved in commented out code. Do we need to have it in the repository?
```
# save(list = "opaka_dat_fims", file = file.path(getwd(), "content", "opaka_dat_fims.RData"))
# load(file = here::here("content", "opaka_dat_fims.RData"))
```

kellijohnson-NOAA commented on content/data_files/opaka_length.RDS:
I think that we should rename this opaka_model.RDS and overwrite the old case study.

kellijohnson-NOAA commented on _quarto.yml:
```diff
@@ -47,7 +47,7 @@ website:
       - href: content/SWFSC-sardine.qmd
         text: SWFSC sardine case study
-      - href: content/PIFS-opakapaka.qmd
+      - href: content/PIFSC-opaka-length-M2.qmd
```
I would like to change this to PIFSC-opakapaka.qmd and remove the old qmd file in favor of the new length-based qmd file. Just one version of opakapaka going forward is best I think.

kellijohnson-NOAA commented on content/PIFSC-opaka-length_M2.qmd:
````diff
@@ -0,0 +1,461 @@
+```{r fims1, warning=FALSE, message=FALSE}
+# automatically loads fims Rcpp module
+#remotes::install_github("NOAA-FIMS/FIMS", ref = "main")
+library(FIMS)
+library(TMB)
+library(dplyr)
+library(tidyr)
````
All of these except r4ss are called in using setup.qmd so they are not needed here.

kellijohnson-NOAA commented on content/PIFSC-opaka-length_M2.qmd:
````diff
@@ -0,0 +1,461 @@
+
+## Setting up Data
+```{r fims-dims}
+load(here::here("content", "data_files", "opaka_length.RDS"))
+#load(file.path(getwd(), "content", "data_files", "opaka_model.RDS"))
````
If the file is renamed then we can just use line 40 and delete line 39.

kellijohnson-NOAA commented on content/PIFSC-opaka-length_M2.qmd:
```diff
@@ -0,0 +1,461 @@
+  # dplyr::mutate(name = "fleet1") |> 
+  # dplyr::bind_rows(opaka_dat_fims) |>
+  dplyr::bind_rows(weight_at_age_data)
+# save(list = "opaka_dat_fims", file = file.path(getwd(), "content", "opaka_dat_fims.RData"))
+# load(file = here::here("content", "opaka_dat_fims.RData"))
```
Delete?

kellijohnson-NOAA commented on content/PIFSC-opaka-length_M2.qmd:
Kyle has an example of plotting selectivity in the scamp case study.

MOshima-PIFSC commented:
@kellijohnson-NOAA I addressed all of your comments and removed some extra, commented out code. The case study should be cleaner now and just one opaka data file and quarto doc.
